File: Web/src/app/models/access-control/asset-permission.ts
import { Asset } from "./asset";
import { Role } from "./role";


export class AssetPermission{
    AuthorizedBy: string;
    AuthorizedDate: Date;
    AssetId: number;
    Asset: Asset;
    RoleId: number;
    Role: Role;

  constructor(ob: any) {
    this.AuthorizedBy = ob.AuthorizedBy;
    this.AuthorizedDate = ob.AuthorizedDate;
    this.AssetId = ob.assetId;
    this.Asset = ob.Asset;
    this.RoleId = ob.RoleId;
    this.Role = ob.Role;
  }
}

File: Api/Timesheet/Entities/Accesscontrol/AssetPermission.cs
namespace Timesheet.Entities.Accesscontrol

{
    public class AssetPermission : EntityBase
    {
        public int AssetId { get; set; }
        public virtual Asset? Asset { get; set; }
        public int EntityId { get; set; }
        public int RoleId { get; set; }
        public virtual Role? Role { get; set; }
    }
}
